kafe2/fit/_base/cost.py

```diff
@@ -407,64 +371,73 @@ def chi2_pointwise_errors(data, model, total_error, parameter_values, parameter_
         :param total_error: total error vector :math:`{\bf \sigma}`
-        :param parameter_values: vector of parameters :math:`{\bf p}`
-        :param parameter_constraints: list of fit parameter constraints
 
         :return: cost function value
         """
-        return _generic_chi2(data=data, model=model, cov_mat_inverse=None, err=total_error, fail_on_zero_errors=True,
-                             parameter_values=parameter_values, parameter_constraints=parameter_constraints)
-
-    @staticmethod
-    def chi2_covariance_fallback(data, model, total_cov_mat_inverse, parameter_values, parameter_constraints):
-        return _generic_chi2(
-            data=data, model=model, cov_mat_inverse=total_cov_mat_inverse, parameter_values=parameter_values,
-            parameter_constraints=parameter_constraints, fail_on_no_matrix=False
-        )
-
-    @staticmethod
-    def chi2_pointwise_errors_fallback(data, model, total_error, parameter_values, parameter_constraints):
-        return _generic_chi2(data=data, model=model, cov_mat_inverse=None, err=total_error, fail_on_zero_errors=False,
-                             parameter_values=parameter_values, parameter_constraints=parameter_constraints)
+        return self._chi2(data=data, model=model, err=total_error)
 
 
-class CostFunctionBase_NegLogLikelihood(CostFunctionBase):
-    def __init__(self, data_point_distribution='poisson'):
+class CostFunction_NegLogLikelihood(CostFunction):
+    def __init__(self, data_point_distribution='poisson', ratio=False):
         r"""
         Base class for built-in negative log-likelihood cost function.
 
         In addition to the measurement data and model predictions, likelihood-fits require a
         probability distribution describing how the measurements are distributed around the model
         predictions.
-        This built-in cost function supports two such distributions: the *Poisson* and *Gaussian* (normal)
-        distributions.
+        This built-in cost function supports two such distributions: the *Poisson* and *Gaussian*
+        (normal) distributions.
 
-        In general, a negative log-likelihood cost function is defined as the double negative logarithm of the
-        product of the individual likelihoods of the data points.
+        In general, a negative log-likelihood cost function is defined as the double negative
+        logarithm of the product of the individual likelihoods of the data points.
 
-        :param data_point_distribution: which type of statistics to use for modelling the distribution of individual data points
+        The likelihood ratio is defined as ratio of the likelihood function for each individual
+        observation, divided by the so-called *marginal likelihood*.
+
+        :param data_point_distribution: which type of statistics to use for modelling the
+        distribution of individual data points
         :type data_point_distribution: ``'poisson'`` or ``'gaussian'``
```

Comment:
For better compatibility for Editors with typee checking like Pycharm
```suggestion
        :param data_point_distribution: Which type of statistics to use for modelling the
            distribution of individual data points. Either ``'poisson'`` or ``'gaussian'``.
        :type data_point_distribution: str
```